components/script/dom/readablestream.rs

```diff
@@ -587,6 +644,110 @@ impl ReadableStream {
             },
         }
     }
+    /// <https://streams.spec.whatwg.org/#abstract-opdef-readablestreamdefaulttee>
+    fn default_tee(&self, clone_for_branch2: bool) -> Fallible<Vec<DomRoot<ReadableStream>>> {
+        // Assert: stream implements ReadableStream.
+
+        // Assert: cloneForBranch2 is a boolean.
+        let clone_for_branch2 = Rc::new(Cell::new(clone_for_branch2));
+
+        // Let reader be ? AcquireReadableStreamDefaultReader(stream).
+        let reader = self.acquire_default_reader(CanGc::note())?;
+
+        // Let reading be false.
+        let reading = Rc::new(Cell::new(false));
+        // Let readAgain be false.
+        let read_again = Rc::new(Cell::new(false));
+        // Let canceled1 be false.
+        let canceled1 = Rc::new(Cell::new(false));
+        // Let canceled2 be false.
+        let canceled2 = Rc::new(Cell::new(false));
+        // // Let reason1 be undefined.
+        let reason1 = Rc::new(Heap::default());
+        // // Let reason2 be undefined.
+        let reason2 = Rc::new(Heap::default());
+        // // Let branch1 be undefined.
+        let mut branch1: Option<DomRoot<ReadableStream>> = None;
+        // // Let branch2 be undefined.
+        let mut branch2: Option<DomRoot<ReadableStream>> = None;
```

Comment:
So everything that is stored on a struct should be `Dom`, which can be created from dom roots, and from which dom roots can be created when needed.